sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/data/EventData.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk
+package trace.data
+
+import cats.Hash
+import cats.Show
+import cats.syntax.show._
+import org.typelevel.otel4s.Attribute
+import org.typelevel.otel4s.semconv.trace.attributes.SemanticAttributes
+
+import java.io.PrintWriter
+import java.io.StringWriter
+import scala.concurrent.duration.FiniteDuration
+
+/** Data representation of an event.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/trace/api/#add-events]]
+  */
+sealed trait EventData {
+
+  /** The name of the event.
+    */
+  def name: String
+
+  /** The epoch time of the event.
+    */
+  def epoch: FiniteDuration
```

Comment:
Would `timestamp` be a better name? That's how its referred to in the spec.

```suggestion
  def timestamp: FiniteDuration
```